tests/rule_based_profiler/alpha/test_profiler_user_workflows.py

```diff
@@ -81,13 +85,19 @@ def test_alice_profiler_user_workflow_single_batch(
     yaml_config: str = alice_columnar_table_single_batch["profiler_config"]
 
     # Instantiate Profiler
-    profiler_config: dict = yaml.load(yaml_config)
+    profiler_config: CommentedMap = yaml.load(yaml_config)
+
+    # Roundtrip through schema validation to add any missing fields
+    deserialized_config: dict = ruleBasedProfilerConfigSchema.load(profiler_config)
```

Comment:
isn't this a config object and not a dict?

Reply:
On second thought I now realize this is indeed a dict because we don't use `NotNullSchema`. Keeping this open because I'd like to discuss the merits of `Schema` vs `NotNullSchema`

Reply:
@cdkini Importantly -- keeping consistent with the objects that read from file -- they inherit from `Schema` are do not have a post-load to return the configuration objects.  Their constituents do.  This is in the docstring now.  Please see above.

Reply:
Sounds good! If we still want to `@post_load` into a config object, we need to ensure we're defining that method at the class level (since we no longer inherit that behavior from `NotNullSchema`)